src/transformer_util.ts

```diff
@@ -392,13 +395,17 @@ function visitNodeStatementsWithSynthesizedComments<T extends ts.Node>(
   const leading = synthesizeDetachedLeadingComments(sourceFile, node, statements);
   const trailing = synthesizeDetachedTrailingComments(sourceFile, node, statements);
   if (leading.commentStmt || trailing.commentStmt) {
-    statements = ts.setTextRange(ts.createNodeArray(statements), {pos: -1, end: -1});
-    if (leading.commentStmt) {
-      statements.unshift(leading.commentStmt);
-    }
-    if (trailing.commentStmt) {
-      statements.push(trailing.commentStmt);
-    }
+    const newStatements: ts.Statement[] =
+        [...arrayOf(leading.commentStmt), ...statements, ...arrayOf(trailing.commentStmt)];
+    statements = ts.setTextRange(ts.createNodeArray(newStatements), {pos: -1, end: -1});
+
+    // The visitor creates a new node with the new statements. However, doing so
+    // reveals a TypeScript bug.
```

Comment:
reference the bug/issue, or at least explain what the bug behaviour is here?

Reply:
I paired with @tbosch on this comment, already forgot what exactly was breaking (other than tests failing). Tobias, can you help me expanding on the comment?

Reply:
Paired with Tobias more today. Still not ideal and somewhat mysterious but we narrowed it down to a small repro case.